src/System.Private.CoreLib/shared/System/Text/Unicode/Utf8Utility.WhiteSpace.cs

```diff
@@ -0,0 +1,141 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using Internal.Runtime.CompilerServices;
+
+#pragma warning disable SA1121 // explicitly using type aliases instead of built-in types
+#if BIT64
+using nint = System.Int64;
+using nuint = System.UInt64;
+#else
+using nint = System.Int32;
+using nuint = System.UInt32;
+#endif
+
+namespace System.Text.Unicode
+{
+    internal static partial class Utf8Utility
+    {
+        /// <summary>
+        /// Returns the index in <paramref name="utf8Data"/> where the first non-whitespace character
+        /// appears, or the input length if the data contains only whitespace characters.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
>[MethodImpl(MethodImplOptions.AggressiveInlining)] [](start = 8, length = 50)

is it possible the compiler will not inline this? I am fine keeping the attribute if we think it is possible.